### DISCUSSION TITLE
is sharing transport a good way in async server, like fastapi

### DISCUSSION THREAD

**Discussion by DeoLeung** [encode/httpx#2518] [Q&A] [answered]
I have a backend which basically proxy income http requests to other destination, using fastapi and httpx
```python

transport = AsyncHTTPTransport(limit=limit)

@app.route('/a')
async def to_a(...):
  async with AsyncClient(transport=transport, base_url='http://a', timeout=a_timeout) as c:
    await c.get('/xxxx')  

@app.route('/b')
async def to_b(...):
  async with AsyncClient(transport=transport, base_url='http://b', timeout=b_timeout) as c:
    await c.get('/xxxx')
```

why not using global `AsyncClient` is that we want to leverage the base_url, and something else for the same destination.

my question is, does the reuse of transport sound ok? which means we share all the http request pools and reduce tcp connection creation?

**Comment by lovelydinosaur** [comment, marked as answer]
Hrm. I think you'll find that the transport is closed when the client is closed, so I'm not sure this'll work as expected?

**Reply by DeoLeung** [reply]
> Hrm. I think you'll find that the transport is closed when the client is closed, so I'm not sure this'll work as expected?

that's a point, we will keep the client open during the backend life cycle, so closing won't be a problem. but definitely we will  double check it and change the code a bit not using context management if have to
